src/crypto/CHIPCryptoPALmbedTLS.cpp

```diff
@@ -1252,6 +1252,14 @@ CHIP_ERROR Spake2p_P256_SHA256_HKDF_HMAC::PointIsValid(void * R)
     return CHIP_NO_ERROR;
 }
 
+#if defined(MBEDTLS_X509_CRT_PARSE_C)
+static int mbedTLSCallbackIgnoreCertValidityCheck(void * data, mbedtls_x509_crt * crt, int depth, uint32_t * flags)
+{
+    *flags &= ~(static_cast<uint32_t>(MBEDTLS_X509_BADCERT_EXPIRED | MBEDTLS_X509_BADCERT_FUTURE));
```

Comment:
Could this not parse the mbedTLS cert and use the depth to validate the rule, ratther than ignore the outcome?

Reply:
I'll do it as a follow up: will see if that code should be restructured.

Reply:
Could you file an issue and add `// TODO(#issue): ` here to track this future work?

Reply:
Opened ticker #21621